tutorials/jmsActiveMQ/src/test/java/com/example/jmsActiveMQ/ApplicationTest.java

```diff
@@ -0,0 +1,205 @@
+  private BrokerService broker;
+  
+  // test messages                                                                                                                                
+  private static String[] lines =
+  {
+    "1st line",
+    "2nd line",
+    "3rd line",
+    "4th line",
+    "5th line",
+  };
+  
+  @Test
+  public void testApplication() throws IOException, Exception {
+    try {
+      // delete output file if it exists                                                                                                          
+      File file = new File(FILE_PATH);
+      file.delete();
+      
+      createAMQClient();
+      
+      // write messages to SQS Queue                                                                                                            
+      writeToQueue();
+
+      // run app asynchronously; terminate after results are checked                                                                              
+      LocalMode.Controller lc = asyncRun();
+
+      // check for presence of output file                                                                                                        
+      chkOutput();
+
+      // compare output lines to input                                                                                                            
+      compare();
+      
+      lc.shutdown();
+    } catch (ConstraintViolationException e) {
+      Assert.fail("constraint violations: " + e.getConstraintViolations());
+    }
+  }
+
+  /**
+   * Start JMS broker from here
+   *
+   * @throws Exception
+   */
+  private void startJMSService() throws Exception
+  {
+    broker = new BrokerService();
+    String brokerName = "ActiveMQOutputOperator-broker";
+    broker.setBrokerName(brokerName);
+    broker.getPersistenceAdapter().setDirectory(new File("target/activemq-data/" + 
+        broker.getBrokerName() + '/' + 
+        org.apache.activemq.store.kahadb.KahaDBPersistenceAdapter.class.getSimpleName()).getAbsoluteFile());
+    broker.addConnector("tcp://localhost:61617?broker.persistent=false");
+    broker.getSystemUsage().getStoreUsage().setLimit(1024 * 1024 * 1024);  // 1GB
+    broker.getSystemUsage().getTempUsage().setLimit(100 * 1024 * 1024);    // 100MB
+    broker.setDeleteAllMessagesOnStartup(true);
+    broker.start();
+  }
+  
+  private void createAMQClient() throws Exception 
+  {
+    startJMSService();
+    
+    // Create a ConnectionFactory
+    ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory("vm://localhost");
+
+    // Create a Connection
+    connection = connectionFactory.createConnection();
+    connection.start();
+
+    // Create a Session
+    session = connection.createSession(false, Session.CLIENT_ACKNOWLEDGE);
+
+    // Create the destination (Topic or Queue)
+    Destination destination = session.createQueue(QUEUE_NAME);
+
+    // Create a MessageProducer from the Session to the Topic or Queue
```

Comment:
Suggest rewriting comment to be more descriptive, e.g.
Create producer for queue from session and set its delivery mode.